AK/CircularQueue.h

```diff
@@ -118,17 +118,17 @@ class CircularQueue {
         size_t m_index { 0 };
     };
 
-    ConstIterator begin() const { return ConstIterator(*this, m_head); }
-    ConstIterator end() const { return ConstIterator(*this, size()); }
+    constexpr ConstIterator begin() const { return ConstIterator(*this, m_head); }
+    constexpr ConstIterator end() const { return ConstIterator(*this, size()); }
 
-    size_t head_index() const { return m_head; }
+    constexpr size_t head_index() const { return m_head; }
 
 protected:
-    T* elements() { return reinterpret_cast<T*>(m_storage); }
-    const T* elements() const { return reinterpret_cast<const T*>(m_storage); }
+    constexpr T* elements() { return &m_storage[0]; }
+    constexpr const T* elements() const { return &m_storage[0]; }
 
     friend class ConstIterator;
-    alignas(T) u8 m_storage[sizeof(T) * Capacity];
+    alignas(T) Array<T, sizeof(T) * Capacity> m_storage {};
```

Comment:
This won't work, you're forcing all the elements to be default-constructed here.
The point of keeping it as untyped memory is to have the storage without the initialisation.

Reply:
Thanks @alimpfard. Good call. I didn't think of that. I will have to think more about this.